src/runtime/pyobject.cs

```diff
@@ -257,17 +268,41 @@ public PyObject GetAttr(string name)
 
 
         /// <summary>
-        /// GetAttr Method
+        /// GetAttr Method. Returns fallback value if getting attribute fails for any reason.
         /// </summary>
         /// <remarks>
         /// Returns the named attribute of the Python object, or the given
         /// default object if the attribute access fails.
         /// </remarks>
+        [Obsolete("This method ignores any Python exceptions, and should not be used. Use GetAttrOrElse.")]
         public PyObject GetAttr(string name, PyObject _default)
         {
+            if (name == null) throw new ArgumentNullException(nameof(name));
+
+            IntPtr op = Runtime.PyObject_GetAttrString(obj, name);
+            if (op == IntPtr.Zero)
+            {
+                Runtime.PyErr_Clear();
+                return _default;
+            }
+            return new PyObject(op);
+        }
+
+        /// <summary>
+        /// Gets the specified attribute. If attribute is not found, returns the fallback value.
+        /// </summary>
+        public PyObject GetAttrOrElse(string name, PyObject _default)
```

Comment:
How much would it hurt to make this behaviour the default? We are using `GetAttr` a lot, I think (even in this file).

Reply:
@filmor I don't know, I don't have any legacy code :)
But that would be a breaking change in contract.

Reply:
I decided to take this change out, so the rest could be merged.